packages/shylu/shylu_dd/frosch/src/Tools/FROSch_Tools_def.hpp

```diff
@@ -386,6 +419,74 @@ namespace FROSch {
+        for(unsigned i = 0;i<nodeEle.size();i++){
+          //vertices
+          if(nodeEle[i]<numEnt[0]){
+            dofEle[i*dofsPerNode+j] = nodeEle[i]+j*numEnt[0];
+            dmapEle[i] = nodeEle[i]+j*numEnt[0];
+          }
+          //shortEdges
+          else if (nodeEle[i]>=numEnt[0] && nodeEle[i]<numEnt[1]+numEnt[0]){
+            dofEle[i*dofsPerNode+j] = nodeEle[i]+j*numEnt[1]+(dofsPerNode-1)*numEnt[0];
+            dmapEle[i] = nodeEle[i]+j*numEnt[1]+(dofsPerNode-1)*numEnt[0];
+          }
+          //straightEdges
+          else if (nodeEle[i]<numEnt[2]+numEnt[0]+numEnt[1] && nodeEle[i]>=numEnt[1]+numEnt[0]){
+            dofEle[i*dofsPerNode+j] = nodeEle[i]+j*numEnt[2]+(numEnt[0]*(dofsPerNode-1))+(numEnt[1]*(dofsPerNode-1));
+            dmapEle[i] = nodeEle[i]+j*numEnt[2]+(numEnt[0]*(dofsPerNode-1))+(numEnt[1]*(dofsPerNode-1));
+          }
+          //edges
+          else if (nodeEle[i]<numEnt[3]+numEnt[2]+numEnt[0]+numEnt[1] && nodeEle[i]>=numEnt[1]+numEnt[0]+numEnt[2]){
+            dofEle[i*dofsPerNode+j] = nodeEle[i]+j*numEnt[3]+(numEnt[0]*(dofsPerNode-1))+(numEnt[1]*(dofsPerNode-1))+(numEnt[2]*(dofsPerNode-1));
+            dmapEle[i] = nodeEle[i]+j*numEnt[3]+(numEnt[0]*(dofsPerNode-1))+(numEnt[1]*(dofsPerNode-1))+(numEnt[2]*(dofsPerNode-1));
+          }
+          //faces
+          else if (nodeEle[i]<numEnt[3]+numEnt[2]+numEnt[0]+numEnt[1]+numEnt[4] && nodeEle[i]>=numEnt[1]+numEnt[0]+numEnt[2]+numEnt[3]){
+            dofEle[i*dofsPerNode+j] = nodeEle[i]+j*numEnt[4]+(numEnt[0]*(dofsPerNode-1))+(numEnt[1]*(dofsPerNode-1))+(numEnt[2]*(dofsPerNode-1))+(numEnt[3]*(dofsPerNode-1));
+            dmapEle[i] = nodeEle[i]+j*numEnt[4]+(numEnt[0]*(dofsPerNode-1))+(numEnt[1]*(dofsPerNode-1))+(numEnt[2]*(dofsPerNode-1))+(numEnt[3]*(dofsPerNode-1));
+          } else{
+            if(nodesMap->getComm()->getRank() == 0) std::cout<<"This should never happen\n";
+          }
+        }
+        dofMaps[j] =   Xpetra::MapFactory<LO,GO,NO>::Build(Xpetra::UseTpetra,-1,dmapEle,0,nodesMap->getComm());
+      }
+    } //RGDSW type CoarseOperator
+     else if(partitionType == 2){
+       for(unsigned j = 0;j<dofsPerNode;j++){
+        for(unsigned i = 0;i<nodeEle.size();i++){
+          //roots
+         if(nodeEle[i]<numEnt[5]){
+            dofEle[i*dofsPerNode+j] = nodeEle[i]+j*numEnt[5];
+            dmapEle[i] = nodeEle[i]+j*numEnt[5];
+          }
+          else if (numEnt[6] !=-1 &&nodeEle[i]>=numEnt[5] && nodeEle[i]<numEnt[6]+numEnt[5]){
+            dofEle[i*dofsPerNode+j] = nodeEle[i]+j*numEnt[6]+(dofsPerNode-1)*numEnt[5];
+            dmapEle[i] = nodeEle[i]+j*numEnt[6]+(dofsPerNode-1)*numEnt[5];
+          }
+        }
+        dofMaps[j] =   Xpetra::MapFactory<LO,GO,NO>::Build(Xpetra::UseTpetra,-1,dmapEle,0,nodesMap->getComm());
+      }
+     }
+    Teuchos::RCP<Xpetra::Map<LO,GO,NO> > tmpMap =   Xpetra::MapFactory<LO,GO,NO>::Build(Xpetra::UseTpetra,-1,dofEle,0,nodesMap->getComm());
```

Comment:
Why do you need all these if conditions?

Reply:
I have to differentiate between the type of the coarse node (vertex,edge,face etc.) in order to have to the correct ordering on the coarse level.  I will check if I can remove some of the if conditions.